presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java

```diff
@@ -708,12 +737,20 @@ private MaterializedResultWithPlan executeInternal(Session session, @Language("S
                 return builder.get()::page;
             });
 
+            Plan plan = createPlan(session, sql, warningCollector);
+
             TaskContext taskContext = TestingTaskContext.builder(notificationExecutor, yieldExecutor, session)
                     .setMaxSpillSize(nodeSpillConfig.getMaxSpillPerNode())
                     .setQueryMaxSpillSize(nodeSpillConfig.getQueryMaxSpillPerNode())
+                    .setQueryMaxTotalMemory(getQueryMaxTotalMemoryPerNode(session))
+                    .setTaskPlan(plan.getRoot())
                     .build();
-
-            Plan plan = createPlan(session, sql, warningCollector);
+            taskContext.getQueryContext().setVerboseExceededMemoryLimitErrorsEnabled(isVerboseExceededMemoryLimitErrorsEnabled(session));
+            taskContext.getQueryContext().setHeapDumpOnExceededMemoryLimitEnabled(isHeapDumpOnExceededMemoryLimitEnabled(session));
+            String heapDumpFilePath = Paths.get(
+                    getHeapDumpFileDirectory(session),
+                    format("%s_%s.hprof", session.getQueryId().getId(), taskContext.getTaskId().getStageExecutionId().getStageId().getId())).toString();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getId&#39; expression. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 752. Consider formatting this data with &#39;%d&#39; instead.